### ISSUE TITLE
Request: add notes for hex editor

### ISSUE THREAD

**Issue by n3rdswithgame** [TASVideos/BizHawk#677]
In the hex editor, could it be possible to be able to right click a address or selected addressses to have the option to add a note, then give the noted addresses a highlight color like a frozen address, but a different color (maybe red?), and then hovering over a noted address will show the note when you hover over a link in a web browser. It would make discoveries easier as the ram values and notes are right there.  L. Spyro's MHS has a similar feature where it lets you color code addresses, but not attach a note.

**Comment by vadosnaprimer** [comment]
Can't do it right now, but hopefully in the near future.

**Comment by adelikat** [comment]
I support this idea

**Comment by adelikat** [comment]
Just an idea, what if instead the hex editor highlighted values that had a corresponding ram watch entry.  Then on hover it showed the description of the ram value

**Comment by vadosnaprimer** [comment]
I always dreamed about unified address manager format that could contain all the useful info about addresses and allow using the same address sets across many tools. If the current ram watch file contains enough of this info, it's sane to reuse it in other tool then.

**Comment by adelikat** [comment]
Yeah, a ram watch has everything you would need.  And could be expanded to have new things in the future

**Comment by EnderOnryo** [comment]
I also second this. Being able to designate custom notes and maybe colors for given addresses would be really nice. 

> Just an idea, what if instead the hex editor highlighted values that had a corresponding ram watch entry. Then on hover it showed the description of the ram value

I also really like this, having ram watch mirror onto other aspects of bizhawk would be great, maybe even allowing lua support for designating notes for those that have more complicated scripts where a simple ram watch would not suffice.
